File: BGC_WebApi/BGC_DataAccess/Entities/Member.cs
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using Microsoft.EntityFrameworkCore;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BGC_DataAccess.Entities;

public class Member
{
    public int MemberId { get; set; }
    public string Name { get; set; }
    public string Email { get; set; }
    //public string Password { get; set; }
    //public string Role { get; set; }  

    public ICollection<RpgCampaign> RpgCampaigns { get;set; }
    public ICollection<GameSession> GameSessions { get; set; }

    public ICollection<RpgCampaignRegistration> RpgCampaignRegistrations { get; set; }
    public ICollection<GameSessionRegistration> GameSessionRegistrations { get; set; }
}

public class MemberConfiguration : IEntityTypeConfiguration<Member>
{
    public void Configure(EntityTypeBuilder<Member> builder)
    {
        builder.HasIndex(n => n.Name, "Name");builder.HasIndex(n => n.Email, "Email");
        builder.Property(m => m.Name).IsRequired().HasMaxLength(50);
        builder.Property(m => m.Email).IsRequired().HasMaxLength(50);
        //builder.Property(m => m.Password).IsRequired().HasMaxLength(50);
        //builder.Property(m => m.Role).IsRequired(); 

    }
}

File: BGC_Portal/src/app/api/models/member.ts
/* tslint:disable */
/* eslint-disable */
import { GameSession } from './game-session';
import { GameSessionRegistration } from './game-session-registration';
import { RpgCampaign } from './rpg-campaign';
import { RpgCampaignRegistration } from './rpg-campaign-registration';
export interface Member {
  email?: null | string;
  gameSessionRegistrations?: null | Array<GameSessionRegistration>;
  gameSessions?: null | Array<GameSession>;
  memberId?: number;
  name?: null | string;
  rpgCampaignRegistrations?: null | Array<RpgCampaignRegistration>;
  rpgCampaigns?: null | Array<RpgCampaign>;
}
